### PR TITLE
Bit Mask utility function

### PR DESCRIPTION
**Pull request by fweik**
Description of changes:
 - tuple interface for Utils::{Array, Vector}
 - Extract bit masking utility function to utils
 - Some small code modernization

### CODE REVIEW

File: src/utils/include/utils/mask.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * @tparam Integral An unsigned integral type
+ * @param mask bit mask, if the i-th bit is set, the i-th element
+ *        in @param t is copied to the output, otherwise it is set to zero.
+ * @param t input elements
+ * @return t partially zeroed out according to mask
+ */
+template <class T, class Integral>
+auto mask(Integral mask, T t)
+    -> std::enable_if_t<std::is_unsigned<Integral>::value &&
+                            (8 * sizeof(Integral) >= tuple_size<T>::value),
```
Comment (RudolfWeeber) [inline src/utils/include/utils/mask.hpp]: Where does the 8 come from?

Comment (fweik) [reply src/utils/include/utils/mask.hpp]: bits vs bytes

Comment (fweik) [reply src/utils/include/utils/mask.hpp]: I gave it a name, you were the second person to ask this ^^

**Review by RudolfWeeber — APPROVED** [review]

**Comment by fweik** [comment]
bors r=RudolfWeeber

**Comment by jngrad** [comment]
bors retry

**Pull request merged**